_posts/pl/2021-02-01-wszystko-co-chcielibyscie-wiedziec-o-fontach-ale-baliscie-sie-zapytac.md

```diff
@@ -0,0 +1,104 @@
+- WOFF (Web Open Font Format) – jest to forma TTF, ale z dodaną kompresją i większym wsparciem przeglądarek. Aktualnie jest wspierany przez wszystkie przeglądarki i jest domyślnym wyborem dla używania zewnętrznych fontów w naszych projektach.
+- WOFF2 (Web Open Font Format 2) – jak po nazwie można się domyślić, jest ulepszoną wersją WOFF. Oferuje mniejsze wielkości plików i lepszą wydajność. Aktualnie jest wspierany przez wszystkie przeglądarki za wyjątkiem IE.
+
+Dodając fonty do aplikacji możemy ograniczyć się tylko do WOFF i WOFF2, jeżeli chcemy wpierać tylko aktualne przeglądarki. Oferują one małe rozmiary plików, a przeglądarka dobrze sobie radzi z wyświetlaniem ich na stronie. Jeżeli na urządzeniach, na których nasza aplikacja będzie uruchamiana brakuje dla nich wsparcia, warto zostać przy domyślnych systemowych fontach z uwagi na wydajność aplikacji. Jeżeli jednak chcemy rozszerzać wsparcie na tak wiele urządzeń jak to możliwe i nie boimy się ich wspierania, możemy wtedy skorzystać z plików EOT i TTF/OTF.
+
+## Dołączanie fontów do naszej aplikacji
+
+Na koniec przyjrzyjmy się jest sposobom dołączania naszych fontów do projektu. Są różne możliwości, a kilka z nich, 
+razem z ich zaletami i wadami, przedstawiam poniżej:
+
+**@Font-face**
+
+Najbardziej znany i lubiany sposób definiowania nowych fontów, które chcemy użyć w naszym projekcie. Tworzymy nowy plik i wrzucamy do niego ścieżki do naszych fontów (najczęściej w formacie WOFF i WOFF2). Plik ma strukturę w której definiujemy nazwę i ścieżki do naszych plików:
+```css
+@font-face {
+  font-family: 'NowyFont';
+  src: url('nowyFont.woff2') format('woff2'),
+       url('nowyFont.woff') format('woff'),
+}
+```
+Struktura jest czytelna, więc możemy łatwo ją modyfikować i dodawać kolejne fonty. Aplikacja nie czeka aż fonty zostaną załadowane, więc nie musimy się martwić, że wydłużamy czas ładowania aplikacji. Może to być poręczne, ale powoduje również pewne problemy, ponieważ przez to mamy często do czynienia z 'mrugnięciem fontu', czyli z szybką zmianą wyglądu tekstu. Są różne strategie, radzenia sobie z tym przez przeglądarki. Najczęściej tekst jest na chwilę niewidzialny i jeżeli po pewnym czasie font nie zostanie załadowany, tekst się wyświetla, ale z użytym domyślnym fontem. W momencie kiedy font się załaduje, tekst 'wskakuje' we właściwy font. Czym font więcej waży i czym mamy gorszy transfer może to potrwać dłużej. Wszystkie żądania odbywają się niezależnie, więc ciężko je razem zgrupować, aby mogły załadować się w tym samym momencie.
+
+**font-display**
+
+Na ratunek z pojawiającym się nagle tekstem przychodzi własność font-display: swap, która możemy ustawić w naszym pliku @font-face. Dzięki niemu nie uświadczymy już dwóch mrugnięć (pojawienie się tekstu i zmiana fontu), a jednego: tekst pojawia się od razu i czeka, aż font zostanie załadowany. Zaletą tego rozwiania jest prostota w konfiguracji. Minus jest to, że tak naprawdę nie wiele nam to jeszcze daje.
```

Comment:
literówka
która ==> którą

i dwie kolejne
Minusem jest to, że tak naprawdę niewiele nam to jeszcze daje.